apps/s_client.c

```diff
@@ -3467,4 +3513,25 @@ static int ldap_ExtendedResponse_parse(const char *buf, long rem)
     return ret;
 }
 
+/*
+ * BASE64 encoder: used only for encoding basic proxy authentication credentials
+ */
+static char *base64encode (const void *buf, size_t len)
+{
+    int i;
+    size_t outl;
+    char  *out;
+
+    /* Calculate size of encoded data */
+    outl = (len / 3);
+    if (len % 3 > 0) outl++;
+    outl<<=2;
+    out = app_malloc(outl+1, "base64 encode buffer");
+
+    i = EVP_EncodeBlock((unsigned char*) out, buf, len);
+    assert(i <= (int) outl);
+    i++;	/* otherwise compiler may complain i is unused */
```

Comment:
Not sure this will be enough for all compilers?

Reply:
Fair enough. I only tested with gcc.
I've added a test for i<0, which I don't think can happen in reality.